src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,123 @@
+import { KeyboardEvent, ReactElement, ReactNode, useRef } from "react"
+import { TabList } from "./components/TabList"
+import { TabProps } from "./components/Tab"
+import { TabPanel } from "./components/TabPanel"
+import { Tab } from "./types"
+
+interface TabsProps {
+  children: ReactElement<TabProps> | Array<ReactElement<TabProps>>
+  currentTab: number
+  onTabChange: (index: number) => void
+}
+
+export function Tabs({ children, onTabChange, currentTab }: TabsProps) {
+  // We use a list of refs for tablist nodes so we can focus them programmatically
+  const tabListRefs = useRef<(HTMLLIElement | null)[]>([])
+
+  // Sometimes children is an array; sometimes it's a single node—let's make it an array
+  const childrenArray = Array.isArray(children) ? children : [children]
+
+  // Let's make sure the
+  let currentIndex =
+    currentTab < 0 || currentTab > childrenArray.length - 1 ? 0 : currentTab
+
+  // Importantly, we need to track the original index of the children; we spread props here
+  const childrenWithIndex = childrenArray.map((child, index) => ({
+    ...child.props,
+    index,
+  }))
+
+  // We need to know if the user passes a currentTab that is disabled—we need to fix that
+  const isDisabledCurrent = childrenWithIndex.find(
+    (child) => child.disabled && child.index === currentIndex,
+  )
+
+  // Now we need to create a list of only enabled children
+  const childrenEnabled = childrenWithIndex.filter((child) => !child.disabled)
+  if (isDisabledCurrent) {
+    /**
+     * If we discovered that the currentTab is disabled, we set the new currentTab to
+     * the index of the first enabled tab we have.
+     **/
+    currentIndex = childrenEnabled[0].index
+  }
```

Comment:
Similar to above, do we need to fix that? Or tell the developer that something is clearly wrong?